File: templates/Enterprise-Template/src/typescript/src/dialogs/onboarding/onboardingResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License

import { TurnContext } from "botbuilder";
import { DictionaryRenderer, LanguageTemplateDictionary, TemplateFunction } from "../templateManager/dictionaryRenderer";
import { TemplateManager } from "../templateManager/templateManager";
import * as i18n from "i18n";

export class OnboardingResponses extends TemplateManager {

   // Fields
   public static ResponseIds = {
        EmailPrompt:  "emailPrompt",
        HaveEmailMessage: "haveEmail",
        HaveNameMessage: "haveName",
        HaveLocationMessage: "haveLocation",
        LocationPrompt: "locationPrompt",
        NamePrompt: "namePrompt",
    }

    private static readonly _responseTemplates: LanguageTemplateDictionary = new Map([
        ["default", new Map([
            [OnboardingResponses.ResponseIds.NamePrompt, OnboardingResponses.fromResources("onBoarding.namePrompt")],
            [OnboardingResponses.ResponseIds.HaveNameMessage, async (context: TurnContext, data: any) => {
                const value = i18n.__("onBoarding.haveName");
                return value.replace("{0}", data.name);
            }],
            [OnboardingResponses.ResponseIds.EmailPrompt, OnboardingResponses.fromResources("onBoarding.emailPrompt")],
            [OnboardingResponses.ResponseIds.HaveEmailMessage, async (context: TurnContext, data: any) => {
                const value = i18n.__("onBoarding.haveEmail");
                return value.replace("{0}", data.email);
            }],
            [OnboardingResponses.ResponseIds.LocationPrompt, OnboardingResponses.fromResources("onBoarding.locationPrompt")],
            [OnboardingResponses.ResponseIds.HaveLocationMessage, async (context: TurnContext, data: any) => {
                const value = i18n.__("onBoarding.haveLocation");
                return value.replace("{0}", data.name).replace("{1}", data.location);
            }]
        ])]
    ]);

    private static fromResources(name: string): TemplateFunction {
        return (context: TurnContext, data: any) => Promise.resolve(i18n.__(name));
    }

    constructor() {
        super();
        this.register(new DictionaryRenderer(OnboardingResponses._responseTemplates));
    }
}

File: samples/EnterpriseNotification/VirtualAssistant/Responses/Onboarding/OnboardingResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.TemplateManager;
using Microsoft.Bot.Schema;

namespace VirtualAssistant.Responses.Onboarding
{
    public class OnboardingResponses : TemplateManager
    {
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                {
                    ResponseIds.EmailPrompt,
                    (context, data) =>
                    MessageFactory.Text(
                        text: OnboardingStrings.EMAIL_PROMPT,
                        ssml: OnboardingStrings.EMAIL_PROMPT,
                        inputHint: InputHints.ExpectingInput)
                },
                {
                    ResponseIds.HaveEmailMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: string.Format(OnboardingStrings.HAVE_EMAIL, data.email),
                        ssml: string.Format(OnboardingStrings.HAVE_EMAIL, data.email),
                        inputHint: InputHints.IgnoringInput)
                },
                {
                    ResponseIds.HaveLocationMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: string.Format(OnboardingStrings.HAVE_LOCATION, data.Name, data.Location),
                        ssml: string.Format(OnboardingStrings.HAVE_LOCATION, data.Name, data.Location),
                        inputHint: InputHints.IgnoringInput)
                },
                {
                    ResponseIds.HaveNameMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: string.Format(OnboardingStrings.HAVE_NAME, data.name),
                        ssml: string.Format(OnboardingStrings.HAVE_NAME, data.name),
                        inputHint: InputHints.IgnoringInput)
                },
                {
                    ResponseIds.NamePrompt,
                    (context, data) =>
                    MessageFactory.Text(
                        text: OnboardingStrings.NAME_PROMPT,
                        ssml: OnboardingStrings.NAME_PROMPT,
                        inputHint: InputHints.ExpectingInput)
                },
                {
                    ResponseIds.LocationPrompt,
                    (context, data) =>
                    MessageFactory.Text(
                        text: OnboardingStrings.LOCATION_PROMPT,
                        ssml: OnboardingStrings.LOCATION_PROMPT,
                        inputHint: InputHints.ExpectingInput)
                }
            }
        };

        public OnboardingResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }

        public class ResponseIds
        {
            public const string EmailPrompt = "emailPrompt";
            public const string HaveEmailMessage = "haveEmail";
            public const string HaveNameMessage = "haveName";
            public const string HaveLocationMessage = "haveLocation";
            public const string LocationPrompt = "locationPrompt";
            public const string NamePrompt = "namePrompt";
        }
    }
}
